[Bug] Lint is not run on `androidTest`

###### Reproduction Steps

1. On the latest master, or a branch off the latest master (I am using [0184435](https://github.com/viciousAegis/Anki-Android/commit/01844357aa09a4c3778f30c4d7ea7798bba71f1d) as my latest pull)
2. run a lint check 

Not sure if this is a problem with my local repo, or the remote one; so posting here

###### Expected Result
Check gets completed with no building errors, and all the lint issues are displayed


###### Actual Result
While lint issues are displayed (not sure if all of them are), the following error arises in the Terminal for me:
````
> Task :AnkiDroid:lintAmazonDebug FAILED
/*
Lint issues listed
*/
27 errors, 0 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:lintAmazonDebug'.
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or create a baseline to see only new errors:
  ```
  android {
      lint {
          baseline = file("lint-baseline.xml")
      }
  }
  ```
  
  For more details, see https://developer.android.com/studio/write/lint#snapshot

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
60 actionable tasks: 3 executed, 1 from cache, 56 up-to-date
````
On checking the `Inspect Code` Output for lint errors:
![image](https://user-images.githubusercontent.com/86671025/153274792-1ab11eb5-c1b6-46c8-abdf-805f07a746bb.png)


###### Debug info
```
MacOs 12.0.1
Android Studio Bumblebee | 2021.1.1
Build #AI-211.7628.21.2111.8092744, built on January 19, 2022
Runtime version: 11.0.11+0-b60-7772763 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.0.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: cn.haojiyou.CodeGlance3 (2.1.0), org.jetbrains.kotlin (211-1.6.10-release-923-AS7442.40), org.intellij.plugins.markdown (211.7142.37)
```
###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -29,8 +29,8 @@ on:
       - 'gradle/**'
 
 jobs:
-  lintRelease:
-    name: Lint Release
+  lintDebug:
+    name: Lint Debug
     timeout-minutes: 15
     runs-on: ubuntu-latest
     steps:
@@ -67,10 +67,13 @@ jobs:
           timeout_minutes: 10
           retry_wait_seconds: 60
           max_attempts: 3
-          command: ./gradlew :AnkiDroid:compilePlayReleaseJavaWithJavac compileLint lint-rules:compileTestJava
+          command: ./gradlew :AnkiDroid:compilePlayDebugJavaWithJavac compileLint lint-rules:compileTestJava
 
-      - name: Run Lint Release
-        run: ./gradlew :api:lintRelease :AnkiDroid:lintPlayRelease
+      - name: Run Lint Debug
+        # "lint" is run under the 'Amazon' flavor, so slow in CI
+        # "lintPlayRelease" doesn't test androidTest
+        # "lintPlayDebug" doesn't test the API
+        run: ./gradlew lintPlayDebug :api:lintDebug
 
       - name: Test Lint Rules
         run: ./gradlew lint-rules:test

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/DeckPickerTest.java
@@ -18,6 +18,7 @@
 package com.ichi2.anki;
 
 import android.Manifest;
+import android.annotation.SuppressLint;
 
 import com.ichi2.anki.tests.InstrumentedTest;
 import com.ichi2.anki.testutil.ThreadUtils;
@@ -49,6 +50,7 @@ import static org.junit.Assert.assertThat;
 import static org.junit.Assume.assumeFalse;
 import static org.junit.Assume.assumeTrue;
 
+@SuppressLint("DirectSystemCurrentTimeMillisUsage")
 public class DeckPickerTest {
     @Rule
     public ActivityScenarioRule<DeckPicker> mActivityRule = new ActivityScenarioRule<>(DeckPicker.class);

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.java
@@ -75,9 +75,9 @@ public class NoteEditorTabOrderTest extends NoteEditorTest {
 
 
     protected void sendKeyDownUp(Activity activity, int keyCode) {
-        BaseInputConnection mInputConnection = new BaseInputConnection(activity.getCurrentFocus(), true);
-        mInputConnection.sendKeyEvent(new KeyEvent(KeyEvent.ACTION_DOWN, keyCode));
-        mInputConnection.sendKeyEvent(new KeyEvent(KeyEvent.ACTION_UP, keyCode));
+        BaseInputConnection inputConnection = new BaseInputConnection(activity.getCurrentFocus(), true);
+        inputConnection.sendKeyEvent(new KeyEvent(KeyEvent.ACTION_DOWN, keyCode));
+        inputConnection.sendKeyEvent(new KeyEvent(KeyEvent.ACTION_UP, keyCode));
     }
 
 

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/TestUtils.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/TestUtils.java
@@ -24,7 +24,6 @@ import org.hamcrest.Description;
 import org.hamcrest.Matcher;
 import org.hamcrest.TypeSafeMatcher;
 
-import java.lang.reflect.Method;
 import java.util.Collection;
 
 import androidx.test.espresso.UiController;
@@ -41,7 +40,7 @@ public class TestUtils {
      */
     public static Matcher<View> withIndex(final Matcher<View> matcher, final int index) {
         return new TypeSafeMatcher<View>() {
-            int currentIndex = 0;
+            int mCurrentIndex = 0;
 
 
             @Override
@@ -54,7 +53,7 @@ public class TestUtils {
 
             @Override
             public boolean matchesSafely(View view) {
-                return matcher.matches(view) && currentIndex++ == index;
+                return matcher.matches(view) && mCurrentIndex++ == index;
             }
         };
     }

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java
@@ -18,6 +18,7 @@
 package com.ichi2.anki.tests;
 
 import android.Manifest;
+import android.annotation.SuppressLint;
 import android.content.SharedPreferences;
 
 import androidx.annotation.StringRes;
@@ -56,6 +57,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 @RunWith(androidx.test.ext.junit.runners.AndroidJUnit4.class)
+@SuppressLint("DirectSystemCurrentTimeMillisUsage")
 public class ACRATest extends InstrumentedTest {
 
     @Rule public GrantPermissionRule mRuntimePermissionRule =

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ACRATest.java
@@ -65,7 +65,7 @@ public class ACRATest extends InstrumentedTest {
 
     private AnkiDroidApp mApp = null;
 
-    private final String[] debugLogcatArguments = { "-t", "300", "-v", "long", "ACRA:S"};
+    private final String[] mDebugLogcatArguments = { "-t", "300", "-v", "long", "ACRA:S"};
     //private String[] prodLogcatArguments = { "-t", "100", "-v", "time", "ActivityManager:I", "SQLiteLog:W", AnkiDroidApp.TAG + ":D", "*:S" };
 
 
@@ -112,7 +112,7 @@ public class ACRATest extends InstrumentedTest {
         setAcraConfig("Debug");
         assertArrayEquals("Debug logcat arguments not set correctly",
                 mApp.getAcraCoreConfigBuilder().build().logcatArguments().toArray(),
-                new ImmutableList<>(debugLogcatArguments).toArray());
+                new ImmutableList<>(mDebugLogcatArguments).toArray());
         verifyDebugACRAPreferences();
     }
 

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
@@ -137,7 +137,7 @@ public class ContentProviderTest extends InstrumentedTest {
      */
     @Before
     public void setUp() throws Exception {
-        Log.i(AnkiDroidApp.TAG, "setUp()");
+        Timber.i("setUp()");
         mCreatedNotes = new ArrayList<>();
         final Collection col = getCol();
 
@@ -208,7 +208,7 @@ public class ContentProviderTest extends InstrumentedTest {
      */
     @After
     public void tearDown() throws Exception {
-        Log.i(AnkiDroidApp.TAG, "tearDown()");
+        Timber.i("tearDown()");
         if (!tearDown) {
             return;
         }

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
@@ -29,10 +29,7 @@ import android.net.Uri;
 import androidx.test.rule.GrantPermissionRule;
 import timber.log.Timber;
 
-import android.util.Log;
-
 import com.ichi2.anki.AbstractFlashcardViewer;
-import com.ichi2.anki.AnkiDroidApp;
 import com.ichi2.anki.CollectionHelper;
 import com.ichi2.anki.FlashCardsContract;
 import com.ichi2.anki.exception.ConfirmModSchemaException;
@@ -121,7 +118,7 @@ public class ContentProviderTest extends InstrumentedTest {
     private static final String[] TEST_NOTE_FIELDS = {"dis is za Fr0nt", "Te$t"};
     private static final String TEST_MODEL_CSS = "styleeeee";
     // Whether tear down should be executed. I.e. if set up was not cancelled.
-    private boolean tearDown;
+    private boolean mTearDown;
 
     private int mNumDecksBeforeTest;
     /* initialCapacity set to expected value when the test is written.
@@ -143,7 +140,7 @@ public class ContentProviderTest extends InstrumentedTest {
 
         // We have parameterized the "schedVersion" variable, if we are on an emulator
         // (so it is safe) we will try to run with multiple scheduler versions
-        tearDown = false;
+        mTearDown = false;
         if (InstrumentedTest.isEmulator()) {
             col.changeSchedulerVer(schedVersion);
         } else {
@@ -153,7 +150,7 @@ public class ContentProviderTest extends InstrumentedTest {
                 assumeThat(col.getSched().getName(), is("std2"));
             }
         }
-        tearDown = true;
+        mTearDown = true;
         // Do not teardown if setup was aborted
 
         // Add a new basic model that we use for testing purposes (existing models could potentially be corrupted)
@@ -209,7 +206,7 @@ public class ContentProviderTest extends InstrumentedTest {
     @After
     public void tearDown() throws Exception {
         Timber.i("tearDown()");
-        if (!tearDown) {
+        if (!mTearDown) {
             return;
         }
         final Collection col = getCol();

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.java
@@ -212,7 +212,7 @@ public class ContentProviderTest extends InstrumentedTest {
         final Collection col = getCol();
         // Delete all notes
         List<Long> remnantNotes = col.findNotes("tag:" + TEST_TAG);
-        if (remnantNotes.size() > 0) {
+        if (!remnantNotes.isEmpty()) {
             long[] noteIds = Utils.collection2Array(remnantNotes);
             col.remNotes(noteIds);
             col.save();

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/NotificationChannelTest.java
@@ -20,7 +20,9 @@ import android.Manifest;
 import android.app.NotificationChannel;
 import android.app.NotificationManager;
 import android.content.Context;
+import android.os.Build;
 
+import androidx.annotation.RequiresApi;
 import androidx.test.annotation.UiThreadTest;
 import androidx.test.rule.GrantPermissionRule;
 
@@ -41,6 +43,7 @@ import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
 @RunWith(androidx.test.ext.junit.runners.AndroidJUnit4.class)
+@RequiresApi(Build.VERSION_CODES.O) // getNotificationChannels, NotificationChannel.getId
 public class NotificationChannelTest extends InstrumentedTest {
 
     @Rule

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/DBTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/DBTest.java
@@ -18,15 +18,11 @@
 package com.ichi2.anki.tests.libanki;
 
 import android.Manifest;
-import android.app.Application;
-import android.content.Context;
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteDatabaseCorruptException;
-import android.os.Build;
 
 import com.ichi2.anki.CollectionHelper;
 import com.ichi2.anki.tests.InstrumentedTest;
-import com.ichi2.compat.CompatHelper;
 import com.ichi2.libanki.DB;
 
 import org.junit.Assert;
@@ -39,9 +35,7 @@ import java.io.FileOutputStream;
 import java.util.Random;
 
 import androidx.sqlite.db.SupportSQLiteDatabase;
-import androidx.test.core.app.ApplicationProvider;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
-import androidx.test.platform.app.InstrumentationRegistry;
 import androidx.test.rule.GrantPermissionRule;
 
 @RunWith(AndroidJUnit4.class)
@@ -62,7 +56,7 @@ public class DBTest extends InstrumentedTest {
         Assert.assertFalse("database exists already", illFatedDBFile.exists());
 
         TestDB illFatedDB = new TestDB(illFatedDBFile.getCanonicalPath());
-        Assert.assertFalse("database should not be corrupt yet", illFatedDB.databaseIsCorrupt);
+        Assert.assertFalse("database should not be corrupt yet", illFatedDB.mDatabaseIsCorrupt);
 
         // Scribble in it
         byte[] b = new byte[1024];
@@ -81,7 +75,7 @@ public class DBTest extends InstrumentedTest {
             // do nothing, it is expected
         }
 
-        Assert.assertTrue("database corruption not detected", illFatedDB.databaseIsCorrupt);
+        Assert.assertTrue("database corruption not detected", illFatedDB.mDatabaseIsCorrupt);
 
         // our handler avoids deleting databases, in contrast with default handler
         Assert.assertTrue("database incorrectly deleted on corruption", illFatedDBFile.exists());
@@ -95,7 +89,7 @@ public class DBTest extends InstrumentedTest {
     // Test fixture that lets us inspect corruption handler status
     public static class TestDB extends DB {
 
-        private boolean databaseIsCorrupt = false;
+        private boolean mDatabaseIsCorrupt = false;
 
         private TestDB(String ankiFilename) {
             super(ankiFilename);
@@ -113,7 +107,7 @@ public class DBTest extends InstrumentedTest {
 
             @Override
             public void onCorruption(SupportSQLiteDatabase db) {
-                databaseIsCorrupt = true;
+                mDatabaseIsCorrupt = true;
                 super.onCorruption(db);
             }
         }

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/ImportTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/ImportTest.java
@@ -63,7 +63,7 @@ import static org.junit.Assert.assertTrue;
 @RunWith(AndroidJUnit4.class)
 public class ImportTest extends InstrumentedTest {
 
-    private Collection testCol;
+    private Collection mTestCol;
 
     @Rule
     public GrantPermissionRule mRuntimePermissionRule =
@@ -85,30 +85,30 @@ public class ImportTest extends InstrumentedTest {
 
     @Before
     public void setUp() throws IOException {
-        testCol = getEmptyCol();
+        mTestCol = getEmptyCol();
     }
 
     @After
     public void tearDown() {
-        testCol.close();
+        mTestCol.close();
     }
 
     @Test
     public void testAnki2Mediadupes() throws IOException, JSONException, ImportExportException {
 
         // add a note that references a sound
-        Note n = testCol.newNote();
+        Note n = mTestCol.newNote();
         n.setField(0, "[sound:foo.mp3]");
         long mid = n.model().getLong("id");
-        testCol.addNote(n);
+        mTestCol.addNote(n);
         // add that sound to the media folder
-        FileOutputStream os = new FileOutputStream(new File(testCol.getMedia().dir(), "foo.mp3"), false);
+        FileOutputStream os = new FileOutputStream(new File(mTestCol.getMedia().dir(), "foo.mp3"), false);
         os.write("foo".getBytes());
         os.close();
-        testCol.close();
+        mTestCol.close();
         // it should be imported correctly into an empty deck
         Collection empty = getEmptyCol();
-        Importer imp = new Anki2Importer(empty, testCol.getPath());
+        Importer imp = new Anki2Importer(empty, mTestCol.getPath());
         imp.run();
         List<String> expected = Collections.singletonList("foo.mp3");
         List<String> actual = Arrays.asList(new File(empty.getMedia().dir()).list());
@@ -116,7 +116,7 @@ public class ImportTest extends InstrumentedTest {
         assertEquals(expected.size(), actual.size());
         // and importing again will not duplicate, as the file content matches
         empty.remCards(empty.getDb().queryLongList("select id from cards"));
-        imp = new Anki2Importer(empty, testCol.getPath());
+        imp = new Anki2Importer(empty, mTestCol.getPath());
         imp.run();
         expected = Collections.singletonList("foo.mp3");
         actual = Arrays.asList(new File(empty.getMedia().dir()).list());
@@ -129,7 +129,7 @@ public class ImportTest extends InstrumentedTest {
         os = new FileOutputStream(new File(empty.getMedia().dir(), "foo.mp3"), false);
         os.write("bar".getBytes());
         os.close();
-        imp = new Anki2Importer(empty, testCol.getPath());
+        imp = new Anki2Importer(empty, mTestCol.getPath());
         imp.run();
         expected = Arrays.asList("foo.mp3", String.format("foo_%s.mp3", mid));
         actual = Arrays.asList(new File(empty.getMedia().dir()).list());
@@ -142,7 +142,7 @@ public class ImportTest extends InstrumentedTest {
         os = new FileOutputStream(new File(empty.getMedia().dir(), "foo.mp3"));
         os.write("bar".getBytes());
         os.close();
-        imp = new Anki2Importer(empty, testCol.getPath());
+        imp = new Anki2Importer(empty, mTestCol.getPath());
         imp.run();
         expected =  Arrays.asList("foo.mp3", String.format("foo_%s.mp3", mid));
         actual = Arrays.asList(new File(empty.getMedia().dir()).list());
@@ -157,32 +157,32 @@ public class ImportTest extends InstrumentedTest {
     public void testApkg() throws IOException, ImportExportException {
 
         String apkg = Shared.getTestFilePath(getTestContext(), "media.apkg");
-        Importer imp = new AnkiPackageImporter(testCol, apkg);
+        Importer imp = new AnkiPackageImporter(mTestCol, apkg);
         List<String> expected = Collections.emptyList();
-        List<String> actual = Arrays.asList(new File(testCol.getMedia().dir()).list());
+        List<String> actual = Arrays.asList(new File(mTestCol.getMedia().dir()).list());
         actual.retainAll(expected);
         assertEquals(actual.size(), expected.size());
         imp.run();
         expected = Collections.singletonList("foo.wav");
-        actual = Arrays.asList(new File(testCol.getMedia().dir()).list());
+        actual = Arrays.asList(new File(mTestCol.getMedia().dir()).list());
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
         // import again should be idempotent in terms of media
-        testCol.remCards(testCol.getDb().queryLongList("select id from cards"));
-        imp = new AnkiPackageImporter(testCol, apkg);
+        mTestCol.remCards(mTestCol.getDb().queryLongList("select id from cards"));
+        imp = new AnkiPackageImporter(mTestCol, apkg);
         imp.run();
         expected = Collections.singletonList("foo.wav");
-        actual = Arrays.asList(new File(testCol.getMedia().dir()).list());
+        actual = Arrays.asList(new File(mTestCol.getMedia().dir()).list());
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
         // but if the local file has different data, it will rename
-        testCol.remCards(testCol.getDb().queryLongList("select id from cards"));
-        FileOutputStream os = new FileOutputStream(new File(testCol.getMedia().dir(), "foo.wav"), false);
+        mTestCol.remCards(mTestCol.getDb().queryLongList("select id from cards"));
+        FileOutputStream os = new FileOutputStream(new File(mTestCol.getMedia().dir(), "foo.wav"), false);
         os.write("xyz".getBytes());
         os.close();
-        imp = new AnkiPackageImporter(testCol, apkg);
+        imp = new AnkiPackageImporter(mTestCol, apkg);
         imp.run();
-        assertEquals(2, new File(testCol.getMedia().dir()).list().length);
+        assertEquals(2, new File(mTestCol.getMedia().dir()).list().length);
     }
 
     @Test
@@ -191,54 +191,54 @@ public class ImportTest extends InstrumentedTest {
         // changed, not the number of cards/fields
         // import the first version of the model
         String tmp = Shared.getTestFilePath(getTestContext(), "diffmodeltemplates-1.apkg");
-        AnkiPackageImporter imp = new AnkiPackageImporter(testCol, tmp);
+        AnkiPackageImporter imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.setDupeOnSchemaChange(true);
         imp.run();
         // then the version with updated template
         tmp = Shared.getTestFilePath(getTestContext(), "diffmodeltemplates-2.apkg");
-        imp = new AnkiPackageImporter(testCol, tmp);
+        imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.setDupeOnSchemaChange(true);
         imp.run();
         // collection should contain the note we imported
-        assertEquals(1, testCol.noteCount());
+        assertEquals(1, mTestCol.noteCount());
         // the front template should contain the text added in the 2nd package
-        Long tcid = testCol.findCards("").get(0);
-        Note tnote = testCol.getCard(tcid).note();
-        assertTrue(testCol.findTemplates(tnote).get(0).getString("qfmt").contains("Changed Front Template"));
+        Long tcid = mTestCol.findCards("").get(0);
+        Note tnote = mTestCol.getCard(tcid).note();
+        assertTrue(mTestCol.findTemplates(tnote).get(0).getString("qfmt").contains("Changed Front Template"));
     }
 
     @Test
     public void testAnki2Updates() throws IOException, ImportExportException {
         // create a new empty deck
         String tmp = Shared.getTestFilePath(getTestContext(), "update1.apkg");
-        AnkiPackageImporter imp = new AnkiPackageImporter(testCol, tmp);
+        AnkiPackageImporter imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.run();
         assertEquals(0, imp.getDupes());
         assertEquals(1, imp.getAdded());
         assertEquals(0, imp.getUpdated());
         // importing again should be idempotent
-        imp = new AnkiPackageImporter(testCol, tmp);
+        imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.run();
         assertEquals(1, imp.getDupes());
         assertEquals(0, imp.getAdded());
         assertEquals(0, imp.getUpdated());
         // importing a newer note should update
-        assertEquals(1, testCol.noteCount());
-        assertTrue(testCol.getDb().queryString("select flds from notes").startsWith("hello"));
+        assertEquals(1, mTestCol.noteCount());
+        assertTrue(mTestCol.getDb().queryString("select flds from notes").startsWith("hello"));
         tmp = Shared.getTestFilePath(getTestContext(), "update2.apkg");
-        imp = new AnkiPackageImporter(testCol, tmp);
+        imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.run();
         assertEquals(1, imp.getDupes());
         assertEquals(0, imp.getAdded());
         assertEquals(1, imp.getUpdated());
-        assertTrue(testCol.getDb().queryString("select flds from notes").startsWith("goodbye"));
+        assertTrue(mTestCol.getDb().queryString("select flds from notes").startsWith("goodbye"));
     }
 
     @Test
     @SdkSuppress(minSdkVersion = Build.VERSION_CODES.O)
     public void testCsv() throws IOException {
         String file = Shared.getTestFilePath(getTestContext(), "text-2fields.txt");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.initMapping();
         i.run();
         if (TestEnvironment.isDisplayingDefaultEnglishStrings()) {
@@ -272,7 +272,7 @@ public class ImportTest extends InstrumentedTest {
         }
         assertEquals(5, i.getTotal());
         // but importing should not clobber tags if they're unmapped
-        Note n = testCol.getNote(testCol.getDb().queryLongScalar("select id from notes"));
+        Note n = mTestCol.getNote(mTestCol.getDb().queryLongScalar("select id from notes"));
         n.addTag("test");
         n.flush();
         i.run();
@@ -284,30 +284,30 @@ public class ImportTest extends InstrumentedTest {
         i.run();
         assertEquals(0, i.getTotal());
         // and if dupes mode, will reimport everything
-        assertEquals(5, testCol.cardCount());
+        assertEquals(5, mTestCol.cardCount());
         i.setImportMode(NoteImporter.ImportMode.ADD_MODE);
         i.run();
         // includes repeated field
         assertEquals(6, i.getTotal());
-        assertEquals(11, testCol.cardCount());
+        assertEquals(11, mTestCol.cardCount());
     }
 
     @Test
     @SdkSuppress(minSdkVersion = Build.VERSION_CODES.O)
     public void testCsv2() throws  IOException, ConfirmModSchemaException {
-        ModelManager mm = testCol.getModels();
+        ModelManager mm = mTestCol.getModels();
         Model m = mm.current();
         JSONObject f = mm.newField("Three");
         mm.addField(m, f);
         mm.save(m);
-        Note n = testCol.newNote();
+        Note n = mTestCol.newNote();
         n.setField(0, "1");
         n.setField(1, "2");
         n.setField(2, "3");
-        testCol.addNote(n);
+        mTestCol.addNote(n);
         // an update with unmapped fields should not clobber those fields
         String file = Shared.getTestFilePath(getTestContext(), "text-update.txt");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.initMapping();
         i.run();
         n.load();
@@ -319,10 +319,10 @@ public class ImportTest extends InstrumentedTest {
     @SdkSuppress(minSdkVersion = Build.VERSION_CODES.O)
     public void testCsvWithByteOrderMark() throws IOException {
         String file = Shared.getTestFilePath(getTestContext(), "text-utf8-bom.txt");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.initMapping();
         i.run();
-        Note n = testCol.getNote(testCol.getDb().queryLongScalar("select id from notes"));
+        Note n = mTestCol.getNote(mTestCol.getDb().queryLongScalar("select id from notes"));
         assertThat(Arrays.asList(n.getFields()), contains("Hello", "world"));
     }
 
@@ -332,10 +332,10 @@ public class ImportTest extends InstrumentedTest {
     @SdkSuppress(minSdkVersion = Build.VERSION_CODES.O)
     public void testUcs2CsvWithByteOrderMark() throws IOException {
         String file = Shared.getTestFilePath(getTestContext(), "text-ucs2-be-bom.txt");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.initMapping();
         i.run();
-        Note n = testCol.getNote(testCol.getDb().queryLongScalar("select id from notes"));
+        Note n = mTestCol.getNote(mTestCol.getDb().queryLongScalar("select id from notes"));
         assertThat(Arrays.asList(n.getFields()), contains("Hello", "world"));
     }
 
@@ -344,11 +344,11 @@ public class ImportTest extends InstrumentedTest {
     public void csvManualBasicExample() throws IOException, ConfirmModSchemaException {
         String file = Shared.getTestFilePath(getTestContext(), "text-anki-manual-csv-single-line.txt");
         addFieldToCurrentModel("Third");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.setAllowHtml(true);
         i.initMapping();
         i.run();
-        Note n = testCol.getNote(testCol.getDb().queryLongScalar("select id from notes"));
+        Note n = mTestCol.getNote(mTestCol.getDb().queryLongScalar("select id from notes"));
         assertThat(Arrays.asList(n.getFields()), contains("foo bar", "bar baz", "baz quux"));
     }
 
@@ -357,11 +357,11 @@ public class ImportTest extends InstrumentedTest {
     @SdkSuppress(minSdkVersion = Build.VERSION_CODES.O)
     public void csvManualLineBreakExample() throws IOException {
         String file = Shared.getTestFilePath(getTestContext(), "text-anki-manual-csv-multi-line.txt");
-        TextImporter i = new TextImporter(testCol, file);
+        TextImporter i = new TextImporter(mTestCol, file);
         i.setAllowHtml(true);
         i.initMapping();
         i.run();
-        Note n = testCol.getNote(testCol.getDb().queryLongScalar("select id from notes"));
+        Note n = mTestCol.getNote(mTestCol.getDb().queryLongScalar("select id from notes"));
         assertThat(Arrays.asList(n.getFields()), contains("hello", "this is\na two line answer"));
     }
 
@@ -373,10 +373,10 @@ public class ImportTest extends InstrumentedTest {
     public void testDupeIgnore() throws IOException, ImportExportException {
         // create a new empty deck
         String tmp = Shared.getTestFilePath(getTestContext(), "update1.apkg");
-        AnkiPackageImporter imp = new AnkiPackageImporter(testCol, tmp);
+        AnkiPackageImporter imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.run();
         tmp = Shared.getTestFilePath(getTestContext(), "update3.apkg");
-        imp = new AnkiPackageImporter(testCol, tmp);
+        imp = new AnkiPackageImporter(mTestCol, tmp);
         imp.run();
         // there is a dupe, but it was ignored
         assertEquals(1, imp.getDupes());
@@ -386,7 +386,7 @@ public class ImportTest extends InstrumentedTest {
 
 
     private void addFieldToCurrentModel(String fieldName) throws ConfirmModSchemaException {
-        ModelManager mm = testCol.getModels();
+        ModelManager mm = mTestCol.getModels();
         Model m = mm.current();
         JSONObject f = mm.newField(fieldName);
         mm.addField(m, f);

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/MediaTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/MediaTest.java
@@ -55,7 +55,7 @@ import static org.junit.Assert.fail;
 @RunWith(AndroidJUnit4.class)
 public class MediaTest extends InstrumentedTest {
 
-    private Collection testCol;
+    private Collection mTestCol;
 
     @Rule
     public GrantPermissionRule mRuntimePermissionRule =
@@ -63,12 +63,12 @@ public class MediaTest extends InstrumentedTest {
 
     @Before
     public void setUp() throws IOException {
-        testCol = getEmptyCol();
+        mTestCol = getEmptyCol();
     }
 
     @After
     public void tearDown() {
-        testCol.close();
+        mTestCol.close();
     }
 
     @Test
@@ -82,15 +82,15 @@ public class MediaTest extends InstrumentedTest {
         os.write("hello".getBytes());
         os.close();
         // new file, should preserve name
-        String r = testCol.getMedia().addFile(path);
+        String r = mTestCol.getMedia().addFile(path);
         assertEquals("foo.jpg", r);
         // adding the same file again should not create a duplicate
-        assertEquals("foo.jpg", testCol.getMedia().addFile(path));
+        assertEquals("foo.jpg", mTestCol.getMedia().addFile(path));
         // but if it has a different md5, it should
         os = new FileOutputStream(path, false);
         os.write("world".getBytes());
         os.close();
-        assertEquals("foo (1).jpg", testCol.getMedia().addFile(path));
+        assertEquals("foo (1).jpg", mTestCol.getMedia().addFile(path));
     }
 
     @Test
@@ -104,7 +104,7 @@ public class MediaTest extends InstrumentedTest {
 
         // new file, should preserve name
         try {
-            testCol.getMedia().addFile(path);
+            mTestCol.getMedia().addFile(path);
             fail("exception should be thrown");
         } catch (EmptyMediaException mediaException) {
             // all good
@@ -114,51 +114,51 @@ public class MediaTest extends InstrumentedTest {
 
     @Test
     public void testStrings() {
-        Long mid = testCol.getModels().getModels().entrySet().iterator().next().getKey();
+        Long mid = mTestCol.getModels().getModels().entrySet().iterator().next().getKey();
 
         List<String> expected = Collections.emptyList();
-        List<String> actual = testCol.getMedia().filesInStr(mid, "aoeu");
+        List<String> actual = mTestCol.getMedia().filesInStr(mid, "aoeu");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Collections.singletonList("foo.jpg");
-        actual = testCol.getMedia().filesInStr(mid, "aoeu<img src='foo.jpg'>ao");
+        actual = mTestCol.getMedia().filesInStr(mid, "aoeu<img src='foo.jpg'>ao");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Arrays.asList("foo.jpg", "bar.jpg");
-        actual = testCol.getMedia().filesInStr(mid, "aoeu<img src='foo.jpg'><img src=\"bar.jpg\">ao");
+        actual = mTestCol.getMedia().filesInStr(mid, "aoeu<img src='foo.jpg'><img src=\"bar.jpg\">ao");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Collections.singletonList("foo.jpg");
-        actual = testCol.getMedia().filesInStr(mid, "aoeu<img src=foo.jpg style=bar>ao");
+        actual = mTestCol.getMedia().filesInStr(mid, "aoeu<img src=foo.jpg style=bar>ao");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Arrays.asList("one", "two");
-        actual = testCol.getMedia().filesInStr(mid, "<img src=one><img src=two>");
+        actual = mTestCol.getMedia().filesInStr(mid, "<img src=one><img src=two>");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Collections.singletonList("foo.jpg");
-        actual = testCol.getMedia().filesInStr(mid, "aoeu<img src=\"foo.jpg\">ao");
+        actual = mTestCol.getMedia().filesInStr(mid, "aoeu<img src=\"foo.jpg\">ao");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Arrays.asList("foo.jpg", "fo");
-        actual = testCol.getMedia().filesInStr(mid, "aoeu<img src=\"foo.jpg\"><img class=yo src=fo>ao");
+        actual = mTestCol.getMedia().filesInStr(mid, "aoeu<img src=\"foo.jpg\"><img class=yo src=fo>ao");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
         expected = Collections.singletonList("foo.mp3");
-        actual = testCol.getMedia().filesInStr(mid, "aou[sound:foo.mp3]aou");
+        actual = mTestCol.getMedia().filesInStr(mid, "aou[sound:foo.mp3]aou");
         actual.retainAll(expected);
         assertEquals(expected.size(), actual.size());
 
-        assertEquals("aoeu", testCol.getMedia().strip("aoeu"));
-        assertEquals("aoeuaoeu", testCol.getMedia().strip("aoeu[sound:foo.mp3]aoeu"));
-        assertEquals("aoeu", testCol.getMedia().strip("a<img src=yo>oeu"));
+        assertEquals("aoeu", mTestCol.getMedia().strip("aoeu"));
+        assertEquals("aoeuaoeu", mTestCol.getMedia().strip("aoeu[sound:foo.mp3]aoeu"));
+        assertEquals("aoeu", mTestCol.getMedia().strip("a<img src=yo>oeu"));
         assertEquals("aoeu", Media.escapeImages("aoeu"));
         assertEquals("<img src='http://foo.com'>", Media.escapeImages("<img src='http://foo.com'>"));
         assertEquals("<img src=\"foo%20bar.jpg\">", Media.escapeImages("<img src=\"foo bar.jpg\">"));
@@ -167,26 +167,26 @@ public class MediaTest extends InstrumentedTest {
     @Test
     public void testDeckIntegration() throws IOException, EmptyMediaException {
         // create a media dir
-        testCol.getMedia().dir();
+        mTestCol.getMedia().dir();
         // Put a file into it
         File file = createNonEmptyFile("fake.png");
-        testCol.getMedia().addFile(file);
+        mTestCol.getMedia().addFile(file);
         // add a note which references it
-        Note f = testCol.newNote();
+        Note f = mTestCol.newNote();
         f.setField(0, "one");
         f.setField(1, "<img src='fake.png'>");
-        testCol.addNote(f);
+        mTestCol.addNote(f);
         // and one which references a non-existent file
-        f = testCol.newNote();
+        f = mTestCol.newNote();
         f.setField(0, "one");
         f.setField(1, "<img src='fake2.png'>");
-        testCol.addNote(f);
+        mTestCol.addNote(f);
         // and add another file which isn't used
-        FileOutputStream os = new FileOutputStream(new File(testCol.getMedia().dir(), "foo.jpg"), false);
+        FileOutputStream os = new FileOutputStream(new File(mTestCol.getMedia().dir(), "foo.jpg"), false);
         os.write("test".getBytes());
         os.close();
         // check media
-        List<List<String>> ret = testCol.getMedia().check();
+        List<List<String>> ret = mTestCol.getMedia().check();
         List<String> expected = Collections.singletonList("fake2.png");
         List<String> actual = ret.get(0);
         actual.retainAll(expected);
@@ -207,39 +207,39 @@ public class MediaTest extends InstrumentedTest {
 
     @Test
     public void testChanges() throws IOException, EmptyMediaException {
-        assertNotNull(testCol.getMedia()._changed());
-        assertEquals(0, added(testCol).size());
-        assertEquals(0, removed(testCol).size());
+        assertNotNull(mTestCol.getMedia()._changed());
+        assertEquals(0, added(mTestCol).size());
+        assertEquals(0, removed(mTestCol).size());
         // add a file
         File dir = getTestDir();
         File path = new File(dir, "foo.jpg");
         FileOutputStream os = new FileOutputStream(path, false);
         os.write("hello".getBytes());
         os.close();
-        path = new File(testCol.getMedia().dir(), testCol.getMedia().addFile(path));
+        path = new File(mTestCol.getMedia().dir(), mTestCol.getMedia().addFile(path));
         // should have been logged
-        testCol.getMedia().findChanges();
-        assertThat(added(testCol).size(), is(greaterThan(0)));
-        assertEquals(0, removed(testCol).size());
+        mTestCol.getMedia().findChanges();
+        assertThat(added(mTestCol).size(), is(greaterThan(0)));
+        assertEquals(0, removed(mTestCol).size());
         // if we modify it, the cache won't notice
         os = new FileOutputStream(path, true);
         os.write("world".getBytes());
         os.close();
-        assertEquals(1, added(testCol).size());
-        assertEquals(0, removed(testCol).size());
+        assertEquals(1, added(mTestCol).size());
+        assertEquals(0, removed(mTestCol).size());
         // but if we add another file, it will
         path = new File(path.getAbsolutePath()+"2");
         os = new FileOutputStream(path, true);
         os.write("yo".getBytes());
         os.close();
-        testCol.getMedia().findChanges(true);
-        assertEquals(2, added(testCol).size());
-        assertEquals(0, removed(testCol).size());
+        mTestCol.getMedia().findChanges(true);
+        assertEquals(2, added(mTestCol).size());
+        assertEquals(0, removed(mTestCol).size());
         // deletions should get noticed too
         assertTrue(path.delete());
-        testCol.getMedia().findChanges(true);
-        assertEquals(1, added(testCol).size());
-        assertEquals(1, removed(testCol).size());
+        mTestCol.getMedia().findChanges(true);
+        assertEquals(1, added(mTestCol).size());
+        assertEquals(1, removed(mTestCol).size());
     }
 
 
@@ -247,10 +247,10 @@ public class MediaTest extends InstrumentedTest {
     public void testIllegal() {
         String aString = "a:b|cd\\e/f\0g*h";
         String good = "abcdefgh";
-        assertEquals(good, testCol.getMedia().stripIllegal(aString));
+        assertEquals(good, mTestCol.getMedia().stripIllegal(aString));
         for (int i = 0; i < aString.length(); i++) {
             char c = aString.charAt(i);
-            boolean bad = testCol.getMedia().hasIllegal("something" + c + "morestring");
+            boolean bad = mTestCol.getMedia().hasIllegal("something" + c + "morestring");
             if (bad) {
                 assertEquals(-1, good.indexOf(c));
             } else {

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/ModelTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/ModelTest.java
@@ -39,22 +39,22 @@ import static org.junit.Assume.assumeTrue;
 @RunWith(AndroidJUnit4.class)
 public class ModelTest extends InstrumentedTest {
 
-    private Collection testCol;
+    private Collection mTestCol;
 
     @Before
     public void setUp() throws IOException {
-        testCol = getEmptyCol();
+        mTestCol = getEmptyCol();
     }
 
     @After
     public void tearDown() {
-        testCol.close();
+        mTestCol.close();
     }
 
     @Test
     public void bigQuery() {
         assumeTrue("This test is flaky on API29, ignoring", Build.VERSION.SDK_INT != Build.VERSION_CODES.Q);
-        ModelManager models = testCol.getModels();
+        ModelManager models = mTestCol.getModels();
         Model model = models.all().get(0);
         final String testString = "test";
         final int size = testString.length() * 1024 * 1024;
@@ -67,7 +67,7 @@ public class ModelTest extends InstrumentedTest {
         // Buf should be more than 4MB, so at least two chunks from database.
         models.flush();
         // Reload models
-        testCol.load();
+        mTestCol.load();
         Model newModel = models.all().get(0);
         assertEquals(newModel, model);
     }

--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/RetryRule.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/libanki/RetryRule.java
@@ -32,7 +32,7 @@ public final class RetryRule implements TestRule {
     /**
      * How many times to try a test
      */
-    private int maxTries = 3;
+    private int mMaxTries = 3;
 
 
     /**
@@ -43,7 +43,7 @@ public final class RetryRule implements TestRule {
         if (i < 1) {
             throw new IllegalArgumentException("iterations < 1: " + i);
         }
-        this.maxTries = i;
+        this.mMaxTries = i;
     }
 
 
@@ -69,7 +69,7 @@ public final class RetryRule implements TestRule {
                 Throwable caughtThrowable = null;
 
                 // implement retry logic here
-                for (int i = 0; i < maxTries; i++) {
+                for (int i = 0; i < mMaxTries; i++) {
                     try {
                         base.evaluate();
                         return;
@@ -79,7 +79,7 @@ public final class RetryRule implements TestRule {
                         t.printStackTrace(System.err);
                     }
                 }
-                System.err.println(description.getDisplayName() + ": giving up after " + maxTries + " failures");
+                System.err.println(description.getDisplayName() + ": giving up after " + mMaxTries + " failures");
                 throw caughtThrowable;
             }
         };

--- a/AnkiDroid/src/androidTest/java/com/ichi2/ui/ActionBarOverflowTest.java
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/ui/ActionBarOverflowTest.java
@@ -1,3 +1,18 @@
+/*
+ *  Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
 package com.ichi2.ui;
 
 import android.content.Context;

